src/vendor_app/app/src/main/kotlin/cz/quanti/android/vendor_app/repository/booklet/impl/BookletRepositoryImpl.kt

```diff
@@ -0,0 +1,114 @@
+package cz.quanti.android.vendor_app.repository.booklet.impl
+
+import cz.quanti.android.vendor_app.repository.VendorAPI
+import cz.quanti.android.vendor_app.repository.booklet.BookletRepository
+import cz.quanti.android.vendor_app.repository.booklet.dao.BookletDao
+import cz.quanti.android.vendor_app.repository.booklet.dto.Booklet
+import cz.quanti.android.vendor_app.repository.booklet.dto.api.BookletApiEntity
+import cz.quanti.android.vendor_app.repository.booklet.dto.api.BookletCodesBody
+import cz.quanti.android.vendor_app.repository.booklet.dto.db.BookletDbEntity
+import io.reactivex.Completable
+import io.reactivex.Single
+
+class BookletRepositoryImpl(
+    private val bookletDao: BookletDao,
+    private val api: VendorAPI
+) : BookletRepository {
+
+    override fun getAllDeactivatedBooklets(): Single<List<Booklet>> {
+        return bookletDao.getAllDeactivated().map { list ->
+            list.map { convert(it) }
+        }
+    }
+
+    override fun getNewlyDeactivatedBooklets(): Single<List<Booklet>> {
+        return bookletDao.getNewlyDeactivated().map { list ->
+            list.map { convert(it) }
+        }
+    }
+
+    override fun saveBooklet(booklet: Booklet): Completable {
+        return Completable.fromCallable { bookletDao.insert(convert(booklet)) }
+    }
+
+    override fun getProtectedBookletsFromServer(): Single<Pair<Int, List<Booklet>>> {
+        return api.getProtectedBooklets().map { response ->
+            var booklets = response.body()
+            if (booklets == null) {
+                booklets = listOf()
+            }
+            Pair(response.code(), booklets.map {
+                convert(it).apply {
+                    this.state = Booklet.STATE_PROTECTED
+                }
+            })
+        }
+    }
+
+    override fun getDeactivatedBookletsFromServer(): Single<Pair<Int, List<Booklet>>> {
```

Comment:
Pro interface bych se raději Pair vyhnul. Vytvoř si data class, ať je hned očividný, co to vrací